### ISSUE TITLE
Not compatible with electron main process

### ISSUE THREAD

**Issue by pvh** [RangerMauve/discovery-swarm-web#2]
Running this in a non-browser window fails quickly because there's no "self" object there. I suspect there are probably other issues running in a node-ish environment as well.

**Comment by RangerMauve** [comment]
Yeah, for electron / etc, I'd like to get this integrated into hyperdiscovery so that the node-related code will be loaded in Electron, and discovery-swarm-web will be used inside windows.
